crates/cast/bin/tx.rs

```diff
@@ -401,3 +429,32 @@ where
         Ok(self)
     }
 }
+
+/// Helper function that tries to decode custom error name and inputs from error payload data.
+async fn decode_execution_revert(data: &RawValue) -> Result<Option<String>> {
+    if let Some(err_data) = data.to_string().replace("\"", "").strip_prefix("0x") {
+        let selector = err_data.get(..8).unwrap_or_default();
+        if let Some(known_error) = SignaturesIdentifier::new(Config::foundry_cache_dir(), false)?
+            .write()
+            .await
+            .identify_error(&hex::decode(selector)?)
+            .await
+        {
+            let mut decoded_error = known_error.name.clone();
+            if !known_error.inputs.is_empty() {
+                decoded_error.push('(');
+                let error = known_error.decode_error(&hex::decode(err_data)?)?;
```

Comment:
you should be able to do write!("({})", ...) with itertools .format(", ")

Reply:
changed in https://github.com/foundry-rs/foundry/pull/9794/commits/e9b5e66eb1af1ea3cc353da10427f12d7504ce94